pyam/utils.py

```diff
@@ -169,20 +169,94 @@ def is_empty(name, s):
             return False
 
         empty_cols = [c for c in df.columns if is_empty(c, df[c])]
-        return df.drop(columns=empty_cols).dropna(axis=0, how="all")
+        df.drop(columns=empty_cols, inplace=True)
+        df.dropna(axis=0, how="all", inplace=True)
+        return df
```

Comment:
`inplace=True` seldom gives you a speed-up with pandas.

https://stackoverflow.com/a/60020384/2873952 .

`reset_index`, `fillna`, `clip` are exceptions, but there is not a good list.
In general, if one needs to copy the whole data to not do the operation inplace, then it might be faster.

`drop` along columns can be done on a view, so it doesn't matter. `dropna` might actually be faster, but probably you have to copy anyway.